ref/src/yppm_driver.f90

```diff
@@ -327,4 +344,79 @@ subroutine write_state(filename)
 
   end subroutine write_state
 
+  !------------------------------------------------------------------
+  ! interpolate_state
+  !
+  ! Increase the database by interpFactor.
+  !------------------------------------------------------------------
+  subroutine interpolate_state(interpFactor)
+
+    integer, intent(in) :: interpFactor
+
+    ! locals
+    ! Define new variables to replace the old variables.
+    ! Once the new variables are defined, they will replace the
+    ! old variables.
+
+    integer new_isd
+    integer new_ied
+    integer new_jsd
+    integer new_jed
+    integer new_js
+    integer new_je
+    real, allocatable :: new_cry(:, :)
+    real, allocatable :: new_q(:, :)
+    real, allocatable :: new_fy2(:, :)
+    real, allocatable :: new_dya(:, :)
+    integer :: odims(4), idims(4)
+
+    odims(1) = isd
+    odims(2) = ied
+    odims(3) = js
+    odims(4) = je + 1
+    call interpolate_allocate(odims, interpFactor, idims)
+    allocate (new_cry(idims(1):idims(2), idims(3):idims(4)))
+    call interpolate_array(cry, odims, new_cry, idims, interpFactor)
+    
+    call interpolate_allocate(odims, interpFactor, idims)
+    allocate (new_fy2(idims(1):idims(2), idims(3):idims(4)))
+    call interpolate_array(fy2, odims, new_fy2, idims, interpFactor)
+    new_js = idims(3)
+    new_je = idims(4)
+
+    odims(1) = isd
+    odims(2) = ied
+    odims(3) = jsd
+    odims(4) = jed
+    call interpolate_allocate(odims, interpFactor, idims)
+    allocate (new_q(idims(1):idims(2), idims(3):idims(4)))
+    call interpolate_array(q, odims, new_q, idims, interpFactor)
+   
+    call interpolate_allocate(odims, interpFactor, idims)
+    allocate (new_dya(idims(1):idims(2), idims(3):idims(4)))
+    call interpolate_array(dya, odims, new_dya, idims, interpFactor)
+    new_isd = idims(1)
+    new_ied = idims(2)
+    new_jsd = idims(3)
+    new_jed = idims(4)
+
+    ! Exchange the old dimensions to the new dimensions.
+    ied = new_ied
+    jed = new_jed
+    je = new_je - 1
+
+    ! Switch the old arrays to the new arrays.
+    call deallocate_state()
+    allocate (cry(new_isd:new_ied, new_js:new_je))
+    allocate (q(new_isd:new_ied, new_jsd:new_jed))
+    allocate (dya(new_isd:new_ied, new_jsd:new_jed))
+    allocate (fy2(new_isd:new_ied, new_js:new_je))
```

Comment:
The `allocate_state` does both deallocate and allocate, and the new dimensions have been saved, so all of these lines can be replaced with a single call to `allocate_state()`

Reply:
Done.